R/check.R

```diff
@@ -205,3 +205,69 @@ coerce_dt <- function(
     return(dt[])
   }
 }
+
+#' @title Check appropriateness of maximum delay
+#'
+#' @description Check if maximum delay specified by the user is long enough and
+#'   raise potential warnings.
+#'
+#' @inheritParams enw_preprocess_data
+#'
+#' @param cum_coverage The aspired percentage of cases that the maximum delay
+#'   should cover. Defaults to 0.8 (80%).
+#'   
+#' @param warn Should a warning be issued if the cumulative case count is
+#'   below `cum_coverage` for the majority of reference dates?
+#'
+#' @return Numeric, the share of reference dates where the cumulative case count
+#'   is below `cum_coverage`.
+#'
+#' @family check
+#' @export
+#' @examples
+#' check_max_delay(germany_covid19_hosp, max_delay = 20, cum_coverage = 0.8)
+check_max_delay <- function(obs, 
+                            max_delay = 20, 
+                            cum_coverage = 0.8,
+                            warn = TRUE) {
+    
+  obs <- coerce_dt(
+    obs, dates = TRUE, required_cols = c("confirm"), copy = TRUE
+  )
+  
+  stopifnot("`cum_coverage` must be between 0 and 1, e.g. 0.8 for 80%." = 
+              cum_coverage>0 & cum_coverage<=1)
+  
+  obs <- enw_add_max_reported(obs, copy = FALSE)
+  
+  # Note that we if we here filter by the user-specified maximum delay, any
+  # warnings obtained would also apply to a modelled, potentially shorter,
+  # maximum delay.
+  obs <- enw_filter_delay(
+    obs, max_delay = max_delay
+  )
+  
+  # filter by earliest observed report date
+  obs <- obs[,
+     .SD[reference_date >= min(report_date) | is.na(reference_date)],
+     by = .group
+  ]
```

Comment:
I adopted this from `enw_add_incidence`. However, I didn't quite like it there, because it seemed like an unexpected side effect that is not directly related to adding incidence/diff_obs to the obs. So I was wondering if we could maybe move this out of `enw_add_incidence`, perhaps into a separate function?

Reply:
Made an issue for it in #305